proxy/core/acceptor/local.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+"""
+    proxy.py
+    ~~~~~~~~
+    ⚡⚡⚡ Fast, Lightweight, Pluggable, TLS interception capable proxy server focused on
+    Network monitoring, controls & Application development, testing, debugging.
+
+    :copyright: (c) 2013-present by Abhinav Singh and contributors.
+    :license: BSD, see LICENSE for more details.
+
+    .. spelling::
+
+        acceptor
+"""
+import os
+import queue
+import socket
+import logging
+import argparse
+import selectors
+import threading
+import multiprocessing.synchronize
+
+from multiprocessing import connection
+from typing import Optional, Tuple, List, Dict
+from typing import Any      # noqa: W0611   pylint: disable=unused-import
+
+from ...common.utils import is_threadless
+
+from ..event import EventQueue
+
+from .executors import ThreadlessPool
+from .work import Work
+
+logger = logging.getLogger(__name__)
+
+
+class LocalExecutor(threading.Thread):
+    """Listens for READ_EVENT over a queue, accepts and initializes work."""
+
+    def __init__(
+            self,
+            idd: int,
+            flags: argparse.Namespace,
+            sock: socket.socket,
+            evq: 'queue.Queue[Any]',
+            executor_queues: List[connection.Connection],
+            executor_pids: List[int],
+            executor_locks: List[multiprocessing.synchronize.Lock],
+            event_queue: Optional[EventQueue] = None,
+    ) -> None:
+        super().__init__()
+        # Index assigned by `AcceptorPool`
+        self.idd = idd
+        self.sock = sock
+        self.evq = evq
+        self.flags = flags
+        self.executor_queues = executor_queues
+        self.executor_pids = executor_pids
+        self.executor_locks = executor_locks
+        self.event_queue = event_queue
+        # Incremented every time work() is called
+        self._total: int = 0
+        self._selector: Optional[selectors.DefaultSelector] = None
+        self._works: Dict[int, Work] = {}
+
+    def run_once(self) -> bool:
+        try:
+            payload = self.evq.get(block=True, timeout=0.1)
+            if isinstance(payload, bool) and payload is False:
+                return True
+            assert isinstance(payload, tuple)
+            conn, addr = payload
+            addr = None if addr == '' else addr
+            self.dispatch(conn, addr)
+        except queue.Empty:
+            pass
+        return False
+
+    def run(self) -> None:
+        self._selector = selectors.DefaultSelector()
+        try:
+            while True:
+                if self.run_once():
```

Comment:
Why don't you wrap just this line with the suppression? This way you'd be able to dedent a whole lot of code. With the lower level of nesting, it's easier to reason about the code (readability/maintainability).